lib/dialects/mssql/connection-manager.js

```diff
@@ -61,6 +61,9 @@ class ConnectionManager extends AbstractConnectionManager {
     try {
       return await new Promise((resolve, reject) => {
         const connection = new this.lib.Connection(connectionConfig);
+        if (connection.state === connection.STATE.INITIALIZED) {
+          connection.connect();
+        }
```

Comment:
@UziTech I don't think you should be checking against `state` here. Just calling `.connect` should be just fine. 🤔

Reply:
Unless the user is using an old version of tedious where calling connect would throw an error.

Reply:
But this Code leads to an unhandled exception if used with msnodesqlv8 instead of tedious.

connection.state is undefined as well as connection.STATE, so it will throw a `cannot read property INITIALIZED of undefined` Error.

Could you please check if connection.STATE is defined before accessing INITIALIZED? Checking if `.connect` exists would not be suitable because msnodesqlv8 does not require the `.connect()` call at all, this is done in the Constructor already.

Reply:
@killerfurbel This code is not supposed to work with `msnodesqlv8` in the first place. It has a completely different API than `tedious`.

Reply:
Really? See https://github.com/TimelordUK/node-sqlserver-v8#use-with-sequelize ... and why does sequelize support the `dialectModule` property in the constructor, if tedious is the only choice?
I agree, tedious is built in, therefore it is possibly the first choice for most users. But it is possible to use others and this code breaks it... I don't think it has a completely different API - it has a specific sequelize connector (https://github.com/TimelordUK/node-sqlserver-v8/blob/master/lib/sequelize/connection.js).

Reply:
So, the `msnodesqlv8` integration for `sequelize` works by re-implementing the `tedious` API, which to me as the `tedious` maintainer, seems extremely questionable.

Reply:
I think the main idea behind `dialectModule` is so you can e.g. use e.g. a forked version of `tedious` or any other connector used by `sequelize`.

The way that `msnodesqlv8` build compatibility for `sequelize` "works", but I'm not sure that is a robust approach. Shouldn't there be a separate dialect for `msnodesqlv8` if that is supposed to be a fully supported option?

Reply:
Sequelize is not designed to work with `msnodesqlv8`

Reply:
> and this code breaks it... I don't think it has a completely different API

If it wants to keep working with sequelize it could add a `STATE` property.

Reply:
I created an issue in the msnodesqlv8 repo: https://github.com/TimelordUK/node-sqlserver-v8/issues/159

However, since you also backported this behavior also to v5, I would strongly vote for a check within sequelize or revert the commit in the v5 branch. I think "removing" a depreciation warning by implementing a breaking change is not reasonable and should only happen in a major release (--> 6.x).

Reply:
You could create a PR to check for `STATE`. That would probably be the best chance of getting it changed.